dotnet-statsig/src/Statsig/Server/ServerDriver.cs

```diff
@@ -30,14 +31,22 @@ public ServerDriver(string serverSecret, StatsigOptions options = null)
             {
                 throw new ArgumentException("serverSecret cannot be empty.", "serverSecret");
             }
-            if (!serverSecret.StartsWith("secret-"))
-            {
-                throw new ArgumentException("Invalid key provided. Please check your Statsig console to get the right server key.", "serverSecret");
-            }
             if (options == null)
             {
                 options = new StatsigOptions();
             }
+            if (!serverSecret.StartsWith("secret-"))
+            {
+                if (options.AdditionalHeaders.Count == 0) 
+                {
+                    // In a multi-server environment, a client key could be used.  
+                    // In that setup additional headers will be provided.
```

Comment:
this is not something that we are opening up to the public right? If so, I'd remove this comment